packages/introspector/tests/integration/graphql/nodes.test.ts

```diff
@@ -0,0 +1,327 @@
+        // Skip if multi-db not supported
+        if (!MULTIDB_SUPPORT) {
+            // eslint-disable-next-line jest/no-disabled-tests, jest/no-jasmine-globals
+            pending();
+            return;
+        }
+
+        const nodeProperties = { first: "testString", second: neo4j.int(42) };
+        const wSession = driver.session({ defaultAccessMode: neo4j.session.WRITE, database: dbName });
+        await wSession.writeTransaction((tx) =>
+            tx.run(
+                "CREATE (:`Test``Label` {strProp: $props.first}) CREATE (:`Test-Label` {singleProp: $props.second})",
+                { props: nodeProperties }
+            )
+        );
+        const bm = wSession.lastBookmark();
+        await wSession.close();
+
+        const typeDefs = await toGraphQLTypeDefs(sessionFactory(bm));
+
+        expect(typeDefs).toMatchInlineSnapshot(`
+            "type Test_Label2 @node(label: \\"Test-Label\\") {
+            	singleProp: BigInt!
+            }
+
+            type Test_Label @node(label: \\"Test\`Label\\") {
+            	strProp: String!
+            }"
+        `);
+
+        expect(() => new Neo4jGraphQL({ typeDefs, driver })).not.toThrow();
+    });
+    test("Should not include properties with ambiguous types", async () => {
+        // Skip if multi-db not supported
+        if (!MULTIDB_SUPPORT) {
+            // eslint-disable-next-line jest/no-disabled-tests, jest/no-jasmine-globals
+            pending();
+            return;
+        }
+
+        const nodeProperties = { str: "testString", int: neo4j.int(42) };
+        const wSession = driver.session({ defaultAccessMode: neo4j.session.WRITE, database: dbName });
+        await wSession.writeTransaction((tx) =>
+            tx.run(
+                `CREATE (:FullNode {amb: $props.str, str: $props.str}) 
+                CREATE (:FullNode {amb: $props.int, str: $props.str})
+                CREATE (:OnlyAmb {amb: $props.str})
+                CREATE (:OnlyAmb {amb: $props.int})
+                `,
+                { props: nodeProperties }
+            )
+        );
+        const bm = wSession.lastBookmark();
+        await wSession.close();
+
+        const typeDefs = await toGraphQLTypeDefs(sessionFactory(bm));
+
+        expect(typeDefs).toMatchInlineSnapshot(`
+            "type FullNode {
+            	str: String!
+            }"
+        `);
+
+        expect(() => new Neo4jGraphQL({ typeDefs, driver })).not.toThrow();
+    });
+    test("Should not include types with no fields", async () => {
+        // Skip if multi-db not supported
+        if (!MULTIDB_SUPPORT) {
+            // eslint-disable-next-line jest/no-disabled-tests, jest/no-jasmine-globals
+            pending();
+            return;
+        }
+
+        const wSession = driver.session({ defaultAccessMode: neo4j.session.WRITE, database: dbName });
+        await wSession.writeTransaction((tx) => tx.run("CREATE (:EmptyNode) CREATE (:FullNode {prop: 1})"));
```

Comment:
How about when an empty node is related to another node? For example, taking the following suggestion:

```suggestion
        await wSession.writeTransaction((tx) => tx.run("CREATE (:EmptyNode)-[:RELATIONSHIP]->(:FullNode {prop: 1})"));
```

This results in an error, but I would really expect it to output:

```graphql
type EmptyNode {
    relationshipFullNodes: [FullNode] @relationship(type: "RELATIONSHIP", direction: OUT)
}

type FullNode {
    prop: BigInt!
    emptyNodesRelationship: [EmptyNode] @relationship(type: "RELATIONSHIP", direction: IN)
}
```

Reply:
That's a valid use case, well spotted! I'll look into it.